src/lib/embeds.ts

```diff
@@ -22,3 +23,25 @@ export function isEmbedByEmbedder(
   }
   return true
 }
+
+export function isVideoView(
+  v: unknown,
+): v is AppBskyEmbedRecordWithMedia.View | AppBskyEmbedVideo.View {
+  return (
+    AppBskyEmbedVideo.isView(v) ||
+    (AppBskyEmbedRecordWithMedia.isView(v) && AppBskyEmbedVideo.isView(v.media))
+  )
+}
+
+export function getVideoEmbed(v: unknown): AppBskyEmbedVideo.View {
+  if (AppBskyEmbedVideo.isView(v)) {
+    return v
+  }
+  if (
+    AppBskyEmbedRecordWithMedia.isView(v) &&
+    AppBskyEmbedVideo.isView(v.media)
+  ) {
+    return v.media
+  }
+  throw new Error(`Invalid video embed ${JSON.stringify(v)}`)
```

Comment:
Can we avoid this? Instead, it would be nice to do some kind of type refinement or to leave just this function, make it nullable, and then check for `null` instead of `is*`. Basically, let's make this codepath impossible.

Reply:
I couldn't get the type refinement approach to work (TypeScript can't seem to understand that `v.media` is definitely a `AppBskyEmbedVideo.View` no matter how explicit I get with the parameter typing), so I went with the null approach instead. There are some conflicts with your refactor PR - I incorporated the change you made but it will still need to be reconciled.